maven_plugin/src/main/java/io/github/chains_project/maven_lockfile/checksum/FileSystemChecksumCalculator.java

```diff
@@ -182,14 +184,14 @@ public String getDefaultChecksumAlgorithm() {
     }
 
     @Override
-    public ResolvedUrl getArtifactResolvedField(Artifact artifact) {
+    public RepositoryInformation getArtifactResolvedField(Artifact artifact) {
         return getResolvedFieldInternal(resolveDependency(artifact, artifactBuildingRequest), artifactBuildingRequest)
-                .orElse(ResolvedUrl.of(""));
+                .orElse(new RepositoryInformation(ResolvedUrl.Unresolved(), RepositoryId.Default()));
     }
 
     @Override
-    public ResolvedUrl getPluginResolvedField(Artifact artifact) {
+    public RepositoryInformation getPluginResolvedField(Artifact artifact) {
         return getResolvedFieldInternal(resolveDependency(artifact, pluginBuildingRequest), pluginBuildingRequest)
-                .orElse(ResolvedUrl.of(""));
+                .orElse(new RepositoryInformation(ResolvedUrl.Unresolved(), RepositoryId.Default()));
```

Comment:
Potential data inconsistency risk - resolved URL is "Unresolved", but the repository ID matches the default for Maven Central.

Would a `RepositoryId.None()` method (returning an empty string) make sense here?

```suggestion
                .orElse(new RepositoryInformation(ResolvedUrl.Unresolved(), RepositoryId.None()));
```

Reply:
Great catch, have updated ^^